### DISCUSSION TITLE
shopt -s has no effect in configuration?

### DISCUSSION THREAD

**Discussion by OJFord** [CyberShadow/aconfmgr#145] [Q&A]
I'm not sure how, because it just seems to be `source`d normally, which works in my testing, but when done through `aconfmgr save` for example:
```sh
shopt -s extglob
for f in some/patt-!(*but-not-this); do
```

errors as though the `shopt` wasn't there:
```
/[...]/.config/aconfmgr/10-test.sh: line 16: syntax error near unexpected token `('
:::: Fatal error! Stack trace:
::::: /[...]/.config/aconfmgr/10-test.sh:1 [source]
::::: /usr/lib/aconfmgr/common.bash:148 [AconfCompileOutput]
::::: /usr/lib/aconfmgr/common.bash:896 [AconfCompile]
::::: /usr/lib/aconfmgr/save.bash:9 [AconfSave]
::::: /usr/lib/aconfmgr/main.bash:185 [Main]
::::: /usr/lib/aconfmgr/main.bash:205 [source]
::::: /usr/bin/aconfmgr:26 [main]
```

and works fine if I edit it into `common.bash` before the `source`:
https://github.com/CyberShadow/aconfmgr/blob/3309c86ad8702705e1b6833d72f221d3b7ff2e0a/src/common.bash#L149

Is it being intentionally blocked somehow? A side-effect of something else and not fixable? Or unintentional and something that could be fixed?

**Comment by CyberShadow** [comment]
That's curious. Have you tried isolating this in a minimal test case? Perhaps with two shell scripts where one sources the other?
